packages/block-directory/src/store/reducer.js

```diff
@@ -33,6 +32,24 @@ export const downloadableBlocks = ( state = {
 				hasPermission: true,
 				isRequestingDownloadableBlocks: false,
 			};
+	}
+	return state;
+};
+
+/**
+ * Reducer managing the installation and deletion of blocks.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export const blockManagement = ( state = {
+	hasPermission: true,
```

Comment:
It looks like `hasPermission` is in both reducers, which means there would be two separate bits of state managing the same thing (`state.downloadableBlocks.hasPermission` and `state. blockManagement.hasPermission`). I realise you maybe thought the state was merged by `combineReducers`, but instead it creates two different properties on the state.

A good way forward would be to split it out into a separate reducer, something like a `canInstallBlocks` reducer which has a simple boolean state. Looks like it'd handle two actions:
- `RECEIVE_DOWNLOADABLE_BLOCKS`
- `SET_INSTALL_BLOCKS_PERMISSION`

Reply:
Ah, I should've read the PR description 🤦‍♂

Reply:
I have moved `hasPermission` into its own reducer. The reducer makes an assumption that we will only have 1 permission type for now. It can be refactored in the future.

Also, we were setting `hasPermission` in the `downloadableBlocks` reducer after the `getDownloadableBlocks` resolver was returned. That doesn't make any sense since we will always call the `hasPermissions` selector/resolver before making that call.

I looked into the `items` list and it's not used anywhere, I removed it.